src/services/test-helpers/mock/mockApi.ts

```diff
@@ -276,6 +289,188 @@ const referendumInfoOfAt = () =>
 		polkadotRegistry.createType('ReferendumInfo');
 	});
 
+/**
+ * ParasService specific constants
+ * The below types and constants use the rococo registry in order to properly
+ * test the ParasService properly accurately
+ */
+const funds = {
+	retiring: rococoRegistry.createType('bool', false),
+	depositor: rococoRegistry.createType(
+		'AccountId',
+		'14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3'
+	),
+	verifier: null,
+	deposit: rococoRegistry.createType('Balance', 100000000000000),
+	raised: rococoRegistry.createType('Balance', 627500000000000),
+	end: rococoRegistry.createType('BlockNumber', 200000),
+	cap: rococoRegistry.createType(
+		'Balance',
+		'0x0000000000000000016345785d8a0000'
+	),
+	lastContribution: rococoRegistry.createType('LastContribution', {
+		preEnding: 6,
+	}),
+	firstSlot: rococoRegistry.createType('LeasePeriod', 13),
+	lastSlot: rococoRegistry.createType('LeasePeriod', 16),
+	trieIndex: rococoRegistry.createType('TrieIndex', 60),
+};
+
+const paraLifeCycleObjectOne = {
+	onboarding: true,
+	parachain: true,
+};
+const paraLifeCycleObjectTwo = {
+	parathread: true,
+	parachain: false,
+};
+const paraId1 = typeFactory.storageKey(199);
+const paraId2 = typeFactory.storageKey(200);
+const paraLifecycleOne = rococoRegistry.createType(
+	'ParaLifecycle',
+	paraLifeCycleObjectOne
+);
+const paraLifecycleTwo = rococoRegistry.createType(
+	'ParaLifecycle',
+	paraLifeCycleObjectTwo
+);
+const accountIdOne = rococoRegistry.createType(
+	'AccountId',
+	'1TYrFCWxwHA5bhiXf6uLvPfG6eEvrzzL7uiPK3Yc6yHLUqc'
+);
+const accountIdTwo = rococoRegistry.createType(
+	'AccountId',
+	'13NXiLYYzVEjXxU3eaZNcrjEX9vPyVDNNpURCzK8Bj9BiCWH'
+);
+const balanceOfOne = rococoRegistry.createType('BalanceOf', 1000000);
+const balanceOfTwo = rococoRegistry.createType('BalanceOf', 2000000);
+
+const fundsEntries = () =>
+	Promise.resolve().then(() => {
+		const optionFundInfo = rococoRegistry.createType('Option<FundInfo>', funds);
+
+		const entries = [
+			[paraId1, optionFundInfo],
+			[paraId2, optionFundInfo],
+		];
+
+		return entries;
+	});
+
+const fundsAt = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('Option<FundInfo>', funds);
+	});
+
+const fundsKeys = () =>
+	Promise.resolve().then(() => {
+		return [paraId1, paraId2];
+	});
+
+const parasLifecyclesEntriesAt = () =>
+	Promise.resolve().then(() => {
+		return [
+			[paraId1, paraLifecycleOne],
+			[paraId2, paraLifecycleTwo],
+		];
+	});
+
+const parasGenesisArgsAt = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('ParaGenesisArgs', { parachain: true });
+	});
+
+const upcomingParasGenesis = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('Option<ParaGenesisArgs>', {
+			parachain: true,
+		});
+	});
+
+const parasLifecyclesAt = () =>
+	Promise.resolve().then(() => {
+		return typeFactory.optionOf(paraLifecycleOne);
+	});
+
+/**
+ * Used for parachain leases
+ */
+const tupleOne = new Tuple(
```

Comment:
I need to extract this call to `Tuple` and have it live in the `TypeFactory`, same for the following ones below.

Reply:
Resolved!